burn-tensor/src/tensor/ops/activation.rs

```diff
@@ -8,6 +8,49 @@ use super::FloatTensor;
 ///
 /// This trait let backend implementations override activation functions for better performance.
 pub trait ActivationOps<B: Backend> {
+    /// Applies the LeakyReLU activation function.
+    ///
+    /// # Arguments
+    ///
+    /// * `tensor` - The tensor.
+    /// * `alpha` - The alpha value that values smaller than 0 are multiplied with.
+    ///
+    /// # Returns
+    ///
+    /// The output tensor.
+    fn leaky_relu<const D: usize>(
+        tensor: FloatTensor<B, D>,
+        alpha: super::FloatElem<B>,
+    ) -> FloatTensor<B, D> {
+        let mask = B::float_lower_elem(tensor.clone(), 0.elem());
+        let scaled_tensor = B::float_mul_scalar(tensor.clone(), alpha.elem());
+
+        // Update the tensor where the values are `> 0` by `tensor * alpha`.
+        B::float_mask_where(tensor, mask, scaled_tensor)
+    }
+
+    /// Applies the LeakyReLU activation function backward.
+    ///
+    /// # Arguments
+    ///
+    /// * `output` - The output tensor.
+    /// * `alpha` - The alpha value that values smaller than 0 are multiplied with.
+    ///
+    /// # Returns
+    ///
+    /// The gradient.
+    fn leaky_relu_backward<const D: usize>(
+        output: FloatTensor<B, D>,
+        grad: FloatTensor<B, D>,
+        alpha: super::FloatElem<B>,
+    ) -> FloatTensor<B, D> {
+        let mask = B::float_lower_elem(output.clone(), 0.elem());
```

Comment:
I feel like this is maybe wrong. This is supposed to be the same mask as during the forward pass, but it's calculated on the output? Maybe I'm wrong though. @louisfd 

I think this will work if the alpha is a positive float, but otherwise we lose the sign information.

Reply:
tagging @louisfd

Reply:
You're right Nathaniel, if alpha<0 this won't work. However I don't think LeakyRelu is ever used with a negative alpha. We _could_ maybe assert that alpha is positive. 
But, since it would be easy to save the input as state instead of the output to cover the general case, I say we do that.

Reply:
I know that other libraries are checking for a negative Alpha and raising an exception if they encounter one. The reasoning for that is that the autodiff could not handle then negative case. I am fine with implementing either one. 

Just to be sure, you mean to implement it like this:
```rust
let mask = B::float_lower_elem(grad.clone(), 0.elem());
```
correct?

Reply:
Hi @KarstenB 
Not exactly, grad doesn't carry that information. For this function it's only a matter of renaming the argument `output` to `input`. Then it's where we call `leaky_relu_backward` that we must pass the input instead of the output. 

I think you haven't done that part yet but it should be easy since you can copy relu. In `crates/burn-autodiff/src/ops/activation.rs` you'll find the relu function, you can copy it and straightforwardly rename `relu` occurrences to `leaky_relu`. 

Then what matters is the `state` variable. At the bottom you'll find `let state = prep.checkpoint(&tensor);` which means the state is actually the input. Above you'll find `B::relu_backward(state, grad)`, so `state` is the input. I realize there's actually a misnaming for relu as well, because the input is given but `relu_backward`'s first arg is named output. It matters less for relu because the input is <0 and the output is ==0, and in relu_backward we use lower or equal. Because relu_backward already takes the input then you won't have to change anything else than the names in the autodiff function.